examples/example-provider-authentication/README.md

```diff
@@ -0,0 +1,45 @@
+# Example Provider Authentication
+
+This folder contains an example on how to implement the Google Workspace Authentication for Terraform without the need to store credentials somewhere.
+
+This example also provides full impersonation, so the actor will be able to maintain super admin resources (f.e. Role assignments).
+
+For other authentication methods please check [Terraform googleworkspace provider documentation](https://registry.terraform.io/providers/hashicorp/googleworkspace/latest/docs#authorization).
+
+The [main.tf](main.tf) contains the terraform resources for the gcloud part (service account creation and roles). Besides that the configuration of Google Workspace Domain-wide delegation still has to be done, see below.
+
+## Full impersonation with short-lived credentials
```

Comment:
I would personally number the steps rather than use sections. Numbering gives you a clear idea that the steps belong together and need to be followed in order.